rrtypes.go

```diff
@@ -49,11 +57,16 @@ type CAA struct {
 }
 
 func (c CAA) RR() RR {
+	data := fmt.Sprintf(`%d %s %q`, c.Flags, c.Tag, c.Value)
+	// Make sure that the zero value is an empty string
+	if data == `0  ""` {
```

Comment:
```suggestion
	data := fmt.Sprintf(`%d %s %q`, c.Flags, c.Tag, c.Value)
	// Make sure that the zero value is an empty string
	if c.Flags == 0 && c.Tag == "" && c.Value == "" {
```
This feels better to me since it doesn't rely on the exact syntactic combination of the `data` variable. But I guess either could work, I dunno. This also feels a little clearer. (And maybe for the other RR types too -- but this is just a nit, I guess it doesn't really matter)

Reply:
No, that seems like a _much_ better way of doing this; good idea!